### PR TITLE
Separate out boolean comparisons form int comparisons in runtime

### LINKED ISSUE #447: Separate out boolean comparisons form int comparisons in runtime
Separate out boolean comparisons form int comparisons and remove 8b5769f.

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose
Separate out boolean comparisons form int comparisons in runtime
Resolves #447

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: runtime/balrt.h [hunk 1]
```diff
@@ -305,6 +305,24 @@ static READONLY inline int64_t taggedFloatCompare(TaggedPtr lhs, TaggedPtr rhs)
...
+    if (lhsVal == rhsVal) {
+        return COMPARE_EQ;
+    }
+    if (lhsVal < rhsVal) {
+        return COMPARE_LT;
+    }
+    if (lhsVal > rhsVal) {
+        return COMPARE_GT;
+    }
+    return COMPARE_UN;
```
Comment (manuranga) [inline runtime/balrt.h:320]: don't think this is correct, int can be negative and still be considered true, and this can never be `COMPARE_UN`.

Comment (manuranga) [reply runtime/balrt.h:320]: disregard this `taggedToBoolean` already &s with `

Comment (manuranga) [inline runtime/balrt.h]: Boolean can't be `COMPARE_UN`, so above `if` should be an `else`.

Comment (heshanpadmasiri) [reply runtime/balrt.h]: Fixed

**Pull request merged**